app/models/adviser.rb

```diff
@@ -63,6 +63,12 @@ def field_order
 
   private
 
+  # All record of what changed is gone by the time we get to the after_commit
+  # hooks, so we need to store any important changes here to be actioned later.
+  def flag_changes_for_after_commit
```

Comment:
Is it worth explicitly calling out that the `#.firm_id_changed?` method will not work? I remember reading a comment similar to this one (maybe even exactly this one) and thinking "oh but surely you can just use ..." (like an idiot). :)